src/dra/5087C.c

```diff
@@ -19,7 +19,56 @@ s32 func_800F087C(u32 chunkX, u32 chunkY) {
     }
 }
 
-INCLUDE_ASM("dra/nonmatchings/5087C", func_800F0940);
+void func_800F0940(void) {
+    s32 temp;
+
+    switch (g_CurrentRoom.bg[0].flags) {
+    case 1:
+        g_CurrentRoom.bg[0].scrollX.i.hi = g_Camera.posX.i.hi;
+        g_CurrentRoom.bg[0].scrollY.i.hi = g_Camera.posY.i.hi;
+        return;
+    case 2:
+        g_CurrentRoom.bg[0].scrollX.i.hi = (g_Camera.posX.i.hi / 2);
+        g_CurrentRoom.bg[0].scrollY.i.hi = (g_Camera.posY.i.hi / 2) + 0x76;
+        return;
+    case 3:
+        g_CurrentRoom.bg[0].scrollX.i.hi = g_Camera.posX.i.hi / 2;
+        g_CurrentRoom.bg[0].scrollY.i.hi = g_Camera.posY.i.hi;
+        return;
+    case 4:
+        g_CurrentRoom.bg[0].scrollX.i.hi = g_Camera.posX.i.hi;
+        g_CurrentRoom.bg[0].scrollY.i.hi = g_Camera.posY.i.hi / 2;
+        if (g_StageId == 0x25) {
+            g_CurrentRoom.bg[0].scrollY.i.hi += 0x80;
+        }
+        return;
+    case 5:
+        g_CurrentRoom.bg[0].scrollX.i.hi = g_Camera.posX.i.hi / 2;
+        g_CurrentRoom.bg[0].scrollY.i.hi = g_Camera.posY.i.hi / 2;
+        if (g_StageId == 0x26) {
+            g_CurrentRoom.bg[0].scrollX.i.hi += 0x80;
+        }
+        return;
+    case 6:
+        g_CurrentRoom.bg[0].scrollX.i.hi = g_Camera.posX.i.hi / 2;
+        temp = (g_Camera.posY.i.hi / 2 - ((g_CurrentRoom.vSize - 1) << 7)) +
+               (g_CurrentRoom.bg[0].h << 7);
+        g_CurrentRoom.bg[0].scrollY.i.hi = temp;
+        if (g_StageId == 0x26) {
```

Comment:
It seems we don't have an enum for this stage https://github.com/Xeeynamo/sotn-decomp/blob/7deb7ccc9fc71add0dc40a6b5c95fe9e9b7da5f0/include/game.h#L242
I guess it's reverse chapel since `STAGE_DAI == 6`